### PR TITLE
WIP: LSTMBlockFusedCell supports Dropout

### LINKED ISSUE #13649: LSTMBlockFusedCell  does not support using  DropoutWrapper
I am trying to use DropoutWrapper with LSTMBlockFusedCell as follows:

```
cell = tf.contrib.rnn.LSTMBlockFusedCell(num_units,forget_bias) 
cell = tf.contrib.rnn.DropoutWrapper(cell,dropout)
```
 I get an exception that the LSTMBlockFusedCell is not an RNNCell
Message: The parameter cell is not a RNNCell. Which is raised form _like_rnncell during DropoutWrapper initialization. 

It is checking for those proprieties on the cell:
""Checks that a given object is an RNNCell by using duck typing."""
   conditions = [hasattr(cell, "output_size"), hasattr(cell, "state_size"),                 hasattr(cell, "zero_state"), callable(cell)]  LSTMBlockFusedCell does not have output_size , state_size or zero_state properties. 

Should LSTMBlockFusedCell  act like RNNCell to allow using various wrappers?

https://stackoverflow.com/questions/46699985/using-tensorflow-dropoutwrapper-with-lstmblockfusedcell

### PR DESCRIPTION
**Pull request by facaiy**
Fix #13649.

The work has not been done yet. However I'm not sure whether the solution is approved, I mean, using DropoutWrapper for inner implementation. So I open the PR early to collect feedback. cc @ebrevdo.

I'll add test case later.

### CODE REVIEW

File: tensorflow/contrib/rnn/python/ops/lstm_ops.py [hunk 1]
```diff
@@ -613,6 +618,18 @@ def __init__(self,
         share weights, but to avoid mistakes we require reuse=True in such
         cases.  By default this is "lstm_cell", for variable-name compatibility
         with `tf.nn.rnn_cell.LSTMCell`.
+      input_keep_prob: unit Tensor or float between 0 and 1, input keep
```
Comment (ebrevdo) [inline tensorflow/contrib/rnn/python/ops/lstm_ops.py]: I'm starting to rethink this.  All of these additional arguments are basically just so you can essentially call tf.nn.dropout on the inputs?  Is that necessary?  Can't the user just call dropout themselves?

Comment (facaiy) [reply tensorflow/contrib/rnn/python/ops/lstm_ops.py]: Yes, I think so. However it might be not easy to implement variational_recurrent correctly or supports complex structure for users by themselves.

File: tensorflow/contrib/rnn/python/ops/lstm_ops.py [hunk 2]
```diff
@@ -623,6 +640,25 @@ def __init__(self,
     # Inputs must be 3-dimensional.
     self.input_spec = base_layer.InputSpec(ndim=3)
 
+    # Dropout parameter.
+    class DummyRNNCell(object):
```
Comment (ebrevdo) [inline tensorflow/contrib/rnn/python/ops/lstm_ops.py]: i really don't like this approach of creating a dummy RNNCell just to get dropout.

how about refactoring the DropoutWrapper so that you can use its underlying helper functions here?

**Review by ebrevdo — CHANGES_REQUESTED** [review]

File: tensorflow/contrib/rnn/python/ops/lstm_ops.py (hunk 1 above)
Comment (facaiy) [reply tensorflow/contrib/rnn/python/ops/lstm_ops.py]: Thanks, I rechecked the implementation of LSTMBlockFusedCell. Perhaps you're right, `tf.nn.dropout` might be good enough. Let's return to #13649 for discussion.

**Comment by facaiy** [comment]
I'll close the PR until we really need it.
